src/test/java/io/fusionauth/http/io/MultipartStreamTest.java

```diff
@@ -25,11 +25,41 @@
 import java.util.Map;
 
 import io.fusionauth.http.FileInfo;
+import io.fusionauth.http.ParseException;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 import static org.testng.Assert.assertEquals;
 
+/**
+ * @author Brian Pontarelli
+ */
 public class MultipartStreamTest {
+  @DataProvider(name = "badBoundary")
+  public Object[][] badBoundary() {
+    return new Object[][]{
+        {"""
+        ------WebKitFormBoundaryTWfMVJErBoLURJIe\r
+        Content-Disposition: form-data; name="foo"\r
+        \r
+        bar----WebKitFormBoundaryTWfMVJErBoLURJIe--"""},
+        {"""
+        ------WebKitFormBoundaryTWfMVJErBoLURJIe\r
+        Content-Disposition: form-data; name="foo"\r
+        \r
+        bar------WebKitFormBoundaryTWfMVJErBoLURJIe--"""}
+    };
+  }
+
+  @Test(dataProvider = "badBoundary")
+  public void bad_boundaryParameter(String boundary) throws IOException {
+    try {
+      MultipartStream stream = new MultipartStream(new ByteArrayInputStream(boundary.getBytes()), "----WebKitFormBoundaryTWfMVJErBoLURJIe".getBytes(), 1024);
+      stream.process(new HashMap<>(), new LinkedList<>());
+    } catch (ParseException e) {
```

Comment:
We could catch this exception with the `@Test` annotation
```
@Test(expectedExceptions = { ParseException.class })
```

Reply:
I like that it asserts the particular message. It could be rewritten to use JUnit:
```java
ParseException e = assertThrows(
  ParseException.class,
  () -> stream.process(new HashMap<>(), new LinkedList<>())
);
assertEquals(e.getMessage(), "Invalid multipart body. Ran out of data while processing.");
```

The only real benefit is narrowing the scope of where the exception is thrown, but it's also more testy to write it this way IMO.

Reply:
yeah, didn't use the built in because we throw that same exception for several reasons, and wanted to ensure we are throwing the one I expect. 

However, I see you can add `expectedExceptionsMessageRegExp` to the `@Test` annotation, I could try using that in conjection with `expectedExceptions` - but maybe same difference at that point. 

I'll test it and see if it looks cleaner.

Reply:
Looks like it worked, will use it.